docs/proposals/vip-1.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: default
+title: Authentication Service API
+parent: Proposals
+permalink: /docs/proposals/vip-1.md
+---
+
+# Venice Improvement Proposal Template
+
+* **Status**: _Under Discussion_
+* **Author(s)**: _Enrico Olivelli_
+* **Pull Request**: _(Link to the main pull request to resolve this VIP)_
+* **Release**: _(Which release include this VIP)_
+
+## Introduction
+
+Currently, Venice doesn't provide a well-defined extensible way to authenticate clients, but it only supports
+TLS based authentication, and it is hard coded.
+This VIP proposes a new API to write plugins to authenticate clients.
+The first follow-up work will be to implement a plugin to support JWT token based authentication.
+
+## Problem Statement 
+
+Venice's services are based on HTTP/REST APIs, and we need a way to authenticate clients using standard
+mechanisms like JWT tokens and OAuth2.
+It means that all the services must perform authentication (and authorization) checks on each request.
+
+Therefore, we need a way to write plugins to perform authentication checks, this way it will be easy to 
+add more and more mechanisms in the future.
+
+Authentication mechanisms vary a lot from each other, and they are not only based on single steps;
+we need to support at least only mechanisms that work well on HTTP and that can be performed in a single step, 
+like basic HTTP Authentication, that needs only to use an HTTP Header
+or TLS based Authentication, that is based on the client certificate exchanged during the TLS handshake.
+
+
+## Scope
+
+1. **What is in scope?**
+
+   - Define an API to write plugins to perform authentication checks.
```

Comment:
Can we add some note (note necessarily here, but somewhere) that this is specific to authentication of Read/Push?

I don't think this would apply to things like nearline produce or davinci local ingestion since that authentication is done directly against the pubsub broker.

Speaking of the pubsub broker (rambling at this point) one thing that we've tried to have set up (but.... haven't been always successful) is trying to come up with some harmony between the authentication/authorization set up of Venice and the underlying pubsub broker.  The notion here is that the Venice controller is able to support API's which abstract what needs to be set up where.  Within LinkedIn, this looks like having the Venice controller set up ACL's on both Venice and Kafka, and both of those services talk to the same authentication service for fetching rules.  This is a very LinkedIn centric view, do you forsee any problems with this setup with what you have in mind with JWT based authenticaiton?